test/ed25519-test.js

```diff
@@ -0,0 +1,137 @@
+    derivations = require('./fixtures/derivation-fixtures');
+    assert.equal(derivations.length, expectedTests);
+  });
+
+  function test_factory(i) {
+    it('can compute correct a and A for secret: ' + i, function() {
+      var test = derivations[i];
+      var secret = utils.toArray(test.secret_hex, 'hex');
+      var derived = ed25519.keyFromSecret(secret);
+      assert.equal(toHex(derived.privBytes()), test.a_hex);
+      var x_recovered = toHex(ed25519.encodeInt(
+                              ed25519.decodePoint(derived.pubBytes()).getX()));
+      assert.equal(x_recovered, test.A_P.x);
+      assert.equal(toHex(derived.pubBytes()), test.A_hex);
+    });
+  }
+
+  for (var i = 0; i < Math.min(expectedTests, MAX_PROGRAMMATIC); i++)
+    test_factory(i);
+});
+
+describe('sign.input ed25519 test vectors', function() {
+  var expectedTests = 1024;
+  var ed25519, lines;
+
+  before(function() {
+    ed25519 = new eddsa('ed25519');
+    lines = fs.readFileSync(__dirname + '/fixtures/sign.input')
+                  .toString()
+                  .split('\n');
+    assert.equal(lines.length, expectedTests + 1 /*blank line*/);
+  });
+
+  function test_factory(i) {
+    it('vector ' + i, function() {
+      var split = lines[i].toUpperCase().split(':');
+      var derived = ed25519.keyFromSecret(split[0].slice(0, 64));
+      var expectedPk = split[0].slice(64);
+      assert.equal(toHex(derived.pubBytes()), expectedPk);
+      var msg = toArray(split[2], 'hex');
+      var sig = toHex(ed25519.sign(msg, derived));
+      var sigR = sig.slice(0, 64);
+      var sigS = sig.slice(64);
+      assert.equal(sigR, split[3].slice(0, 64));
+      assert.equal(sigS, split[3].slice(64, 128));
+      assert(ed25519.verify(msg, sig, derived.pub()));
+      var forged = msg.length === 0 ? [0x78] /*ord('x')*/:
+                   msg.slice(0, msg.length-1).concat(
+                        (msg[(msg.length-1)] + 1) % 256)
+      assert.equal(msg.length || 1, forged.length);
+      assert(!ed25519.verify(forged, sig, derived.pub()));
+    })
+  }
+  for (var i = 0; i < Math.min(expectedTests, MAX_PROGRAMMATIC); i++)
+    test_factory(i);
+});
+
+describe('EDDSA(\'ed25519\')', function() {
+  var ed25519
+
+  before(function() {
+    ed25519 = new eddsa('ed25519');
+  });
+
+  it('has encBytes of 32', function() {
+    assert.equal(32, ed25519.encBytes);
+  });
+
+  it('can sign/verify messages', function() {
+    var secret = toArray(new Array(65).join('0'), 'hex');
+    assert(secret.length == 32);
+    var msg = [0xB, 0xE, 0xE, 0xF];
+    var derived = ed25519.keyFromSecret(secret);
+    var sig = toHex(ed25519.sign(msg, derived));
+
+    var R = "8F1B9A7FDB22BCD2C15D4695B1CE2B063CBFAEC9B00BE360427BAC9533943F6C";
+    var S = "5F0B380FD7F2E43B70AB2FA29F6C6E3FFC1012710E174786814012324BF19B0C";
+
+    assert.equal(sig.slice(0, 64), R);
+    assert.equal(sig.slice(64), S);
+
+    assert(ed25519.verify(msg, sig, derived.pub()));
```

Comment:
Hrmm, Signature would need an eddsa instance to decodePoints, if the
constructor took say the signature bytes in some form to decode r/s.